site/content/blog/2019-01-20-submitting-great-prs.md

```diff
@@ -0,0 +1,64 @@
+
+But this blog post isn't about writing great code, it's about making your pull request a great experience for you and your reviewers.
+
+I still remember my first pull request. I was a co-op student a few days into my first term with a company. This was before the days of GitHub, and for that matter we weren't even using git at the time. I staged my changes and notified my mentor. Without even looking at my code, he sat me down and said something like this, "Jesse, before you ask anyone else to review your code, I want you to review it yourself." He asked me to don the hat of a reviewer, read through my code, make comments just like a real reviewer, and then send those comments back to him.
+
+I am forever grateful to my mentor for shaping my thinking about reviewing code. Having given more than a few reviews since then, let me share some advice, starting with what my mentor taught me:
+
+### 1) Review your own code first.
+
+You be the one to catch styling issues. You be the one to ask, "does this variable name make any sense?" You be the one to identify problem areas and points of confusion. You be the first reviewer.
+
+Now, not everything deserves to be fixed in every pull request. I love to refactor code, but there's a time and a place and fixing a one-line bug isn't usually the time to fix a multi-file, variable naming issue. This leads to my second point:
+
+### 2) Give a great description.
+
+The description on a pull request is your chance to prove to a reviewer that you have the credentials to fix this problem. How? State the problem in your own words:
+
+> This pull request adds new functionality to expose ...
+
+Tell the reviewer why we need to do this:
+
+> With this new functionality, we can fix a longstanding customer issue that ...
+
+You don't need to outline everything your code is doing -- that's what the code is there for -- but this is also your opportunity to point the reviewer in the right direction:
+
+> The trickiest part of this pull request was the `SqlStore` changes, because ...
+
+and also:
+
+> I really wanted to fix the variable names in this file, but it's out of scope for this pull request.
+
+It's amazing how much you can avoid doing by using the phrase, "out of scope!" Keep that in your back pocket :)
+
+Of course, sometimes you have to fix something else before you can get to the issue at hand. Maybe you decide to bite the bullet and fix that variable name problem because it clarifies the feature you're adding. On to my third point:
+
+### 3) Clean up your history.
+
+Everyone has their own development style. Mine usually consists of mashing the keyboard, then typing:
+
+    git commit -m wip
```

Comment:
Maybe surround with a code block, 0/5

Reply:
The indentation here already renders this as a code block, but perhaps you're suggesting to make this explicit?

![image](https://user-images.githubusercontent.com/1023171/51556429-2ba19880-1e48-11e9-84cb-7a33264e3265.png)

Reply:
Oh I didn't realize the indentation automatically does that. Nevermind 👍

Reply:
I would like to keep this consistent with most of the other files and use backticks to mark the codeblock. 1/5